packages/block/types.ts

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+
+export type { VNode, VElement } from '../million/types';
+export type Props = Record<string, any>;
+
+export class Hole {
+  key: string;
+  constructor(key: string) {
+    this.key = key;
+  }
+}
+
+export class AbstractBlock {
+  root?: HTMLElement;
+  edits?: Edit[];
+  el?: HTMLElement;
+  _parent?: HTMLElement | null;
+  props?: Props | null;
+  key?: string;
+  cache?: Map<number, HTMLElement>;
+  patch(_block: AbstractBlock) {}
+  mount(_parent?: HTMLElement, _refNode: Node | null = null) {}
+  move(_block: AbstractBlock | null = null, _refNode: Node | null = null) {}
+  remove() {}
+  toString() {}
+  get parent(): HTMLElement | null | undefined {
+    return this._parent;
+  }
+}
```

Comment:
I think this class can be made `abstract`, along with the methods:

```suggestion
export abstract class AbstractBlock {
  root?: HTMLElement;
  edits?: Edit[];
  el?: HTMLElement;
  _parent?: HTMLElement | null;
  props?: Props | null;
  key?: string;
  cache?: Map<number, HTMLElement>;
  abstract patch(block: AbstractBlock): HTMLElement;
  abstract mount(parent?: HTMLElement, refNode: Node | null): HTMLElement;
  abstract move(block: AbstractBlock | null = null, refNode: Node | null): void;
  abstract remove(): void;
  abstract toString(): string;
  get parent(): HTMLElement | null | undefined {
    return this._parent;
  }
}
```